internal/rbd/rbd_util.go

```diff
@@ -719,7 +719,14 @@ func (rv *rbdVolume) checkImageChainHasFeature(ctx context.Context, feature uint
 		}
 		err = vol.getImageInfo()
 		if err != nil {
-			util.ErrorLog(ctx, "failed to get image info for %s: %s", vol, err)
+			// call to getImageInfo returns the parent name even if the parent
+			// is in the trash, when we try to open the parent image to get its
+			// information it fails because it is already in trash. We should
+			// treat error as nil if the parent is not found.
+			if errors.Is(err, ErrImageNotFound) {
```

Comment:
Just a question, In case Parent image is not found, why isn't this caught in `getImageInfo()` here? https://github.com/ceph/ceph-csi/blob/8d7c66363daa007c9c28085629b123beb197ece4/internal/rbd/rbd_util.go#L1162

Reply:
`parentInfo, err := image.GetParent()`  will return the parent details even if the parent is in trash, and above `ErrNotFound` will get hit only if the parent is not deleted completely. when you try to call `getImageInfo` for parent again as we are checking the images in the chain it will fail with  `ErrImageNotFound` in rv.open() function call

Reply:
Got it! Thanks for explaining :)

Reply:
@Madhu-1 just to confirm, returning the image info even when an image is in trash can be avoided with any options in Ceph side ?

Reply:
the above cannot be avoided as when you ask for parent details using go-ceph `GetParent` function it provides the parent details but it won't return any details like the image is in the trash or not.
It's possible to check the image is in the trash but it will be a costly operator as we need to list all the images in the trash and do a comparison.

Reply:
> the above cannot be avoided as when you ask for parent details using go-ceph `GetParent` function it provides the parent details but it won't return any details like the image is in the trash or not.

Ok.. so looks like its a go ceph limitation then? if thats the case, I believe the image in trash or not is kind of a good enhancement we need on GetParent call or related calls. Because it can help us in various code paths in CSI layer and avoid some logic errors/complexities.  wdyt @Madhu-1 ?

Reply:
can you open an issue in go-ceph and see if it's possible to get this information?

Reply:
> can you open an issue in go-ceph and see if it's possible to get this information?

Sure..!

Reply:
@humblec anything else is required on this PR? if not I would like to get it in.

Reply:
Lets get this in with priority. Thanks :+1: